UnitTests: Add test TestDeleteNodeAffinityKeyFromNode

iamniting opened the pull request:
Add test for deleteNodeAffinityKeyFromNodes in uninstall_reconciler_test.go

Signed-off-by: Nitin Goyal nigoyal@redhat.com

cc: @raghavendra-talur

umangachapagain commented on pkg/controller/storagecluster/uninstall_reconciler_test.go:
```diff
@@ -131,3 +137,78 @@ func TestDeleteStorageClasses(t *testing.T) {
 	}
 }
+
+func TestDeleteNodeAffinityKeyFromNodes(t *testing.T) {
```
use Table Driven approach and do processing steps  in separate functions outside the main test block for better readability.

iamniting replied:
Done

umangachapagain commented on pkg/controller/storagecluster/uninstall_reconciler_test.go:
```diff
@@ -131,3 +137,78 @@ func TestDeleteStorageClasses(t *testing.T) {
+	}
+}
+
+func assertGetStorageClusterEligibleNodes(
```
```suggestion
func getStorageClusterEligibleNodes(
```

umangachapagain replied:
If you need to assert values of  "node", do it after function call.

iamniting replied:
Deleted the func as I have mainly created the func for the assert and we already have a func for getting nodes

obnoxxx commented:
/ok-to-test

jarrpa commented on pkg/controller/storagecluster/uninstall_reconciler_test.go:
```diff
@@ -154,3 +160,89 @@ func assertTestDeleteStorageClasses(t *testing.T, reconciler ReconcileStorageClu
+
+func TestDeleteNodeAffinityKeyFromNodes(t *testing.T) {
+
+	testList := []struct {
+		createUserDefinedNodeAffinityKey bool
+	}{
+		// case1: verify deleteNodeAffinityKeyFromNodes deletes default NodeAffinityKey only
+		{false},
+		// case2: verify deleteNodeAffinityKeyFromNodes does not deletes user defined NodeAffinityKey
+		{true},
+	}
```
As in https://github.com/openshift/ocs-operator/pull/794#discussion_r505915347 , make this a bit more readable.

iamniting replied:
Done

jarrpa commented:
Also, remove the StorageClasses testing commit, that's not part of this PR.

iamniting commented:
> Please remove the StorageClasses testing commit, that's not part of this PR. Also, you have the wrong subject and message on your commit, it should be for "UnitTests: Add test TestDeleteNodeAffinityKeyFromNode".

Actually, I was doing like this, So I don't have to rebase it later. As changes are in one file and there will be merge conflicts later. Anyway, I have removed those changes I will rebase them later itself.
I have changed the commit-msg.

iamniting commented:
Closing it as it is part of https://github.com/openshift/ocs-operator/pull/807
